### PR TITLE
Few opts

### PR DESCRIPTION
**Pull request by PeterTillema**

### CODE REVIEW

File: src/graphx/graphx.asm [hunk 1]
```diff
@@ -460,7 +459,6 @@ _GetPixel:
 	inc	hl
 	inc	hl
 	inc	hl                          ; move to next argument
-	ld	de,0
```
Comment (runer112) [inline src/graphx/graphx.asm]: This removal is not safe. Although _PixelPtr_ASM is documented as the y input being only 8-bit, the full 24-bit value is used for clipping.

File: src/graphx/graphx.asm [hunk 2]
```diff
@@ -481,7 +479,6 @@ _SetPixel:
...
```
Comment (runer112) [inline src/graphx/graphx.asm]: Ditto.

File: src/graphx/graphx.asm [hunk 3]
```diff
@@ -648,12 +645,12 @@ _Rectangle_NoClip:
 ;  arg3 : Height
 ; Returns:
 ;  None
-	ld	iy,0
+	ld	iy,-1
```
Comment (runer112) [inline src/graphx/graphx.asm]: Why..?

File: src/graphx/graphx.asm [hunk 4]
```diff
@@ -679,36 +676,37 @@ _HorizLine:
 ;  arg2 : Length
 ; Returns:
 ;  None
-	ld	iy,0
+	ld	iy,-1
```
Comment (runer112) [inline src/graphx/graphx.asm]: Why..?

File: src/graphx/graphx.asm [hunk 5]
```diff
@@ -5542,8 +5540,8 @@ _PixelPtr_ASM:
 	ld	hl,-lcdWidth
 	add	hl,bc
 	ret	c
-	ld	hl,-lcdHeight
-	add	hl,de
+	ld	a,256-lcdHeight
```
Comment (runer112) [inline src/graphx/graphx.asm]: I believe that some usages expect clipping on the full 24-bit value.

File: src/graphx/graphx.asm [hunk 6]
```diff
@@ -5542,8 +5540,8 @@ _PixelPtr_ASM:
...
+	add	a,e
```
Comment (runer112) [inline src/graphx/graphx.asm]: _Circle calls _SetPixel_ASM, which calls this (_PixelPtr_ASM). Although this routine is documented as y only being 8-bit, _Circle needs it to be 24-bit for proper clipping.

**Comment by mateoconlechuga** [comment]
This is because clipping an 8-bit value when it is already out of range of an 8-bit signed number is not doable.

**Pull request merged**